mobly/controllers/android_device.py

```diff
@@ -812,12 +826,16 @@ def reboot(self):
         if self.is_bootloader:
             self.fastboot.reboot()
             return
+        snippet_info = self._get_active_snippet_info()
         self.stop_services()
         self.adb.reboot()
         self.wait_for_boot_completion()
         if self.is_rootable:
             self.root_adb()
-        self.start_services(skip_sl4a=getattr(self, KEY_SKIP_SL4A, False))
+        skip_sl4a = getattr(self, KEY_SKIP_SL4A, False) or self.sl4a is None
```

Comment:
Curious what was wrong with the old behaviour? Under what circumstances would skip_sl4a be true but self.sl4a be None?